src/redux/slices/transactions/index.js

```diff
@@ -1,7 +1,103 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import set from 'lodash-es/set';
 import { createSelector } from 'reselect';
 
+import { getTransactions, transactionExtraInfo } from '../../../utils/explorer-api';
+
 const SLICE_NAME = 'transactions';
 
+const initialState = {
+    transactions: {
+        byAccountId: {}
+    },
+    status: {
+        
+    }
+};
+
+const fetchTransactions = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactions`,
+    async ({ accountId }, { dispatch, getState }) => {
+        const transactions = await getTransactions({ accountId });
+
+        const { actions: { addTransactions, updateTransactions } } = transactionsSlice;
+
+        !selectTransactionsByAccountId(getState(), { accountId }).length
+            ? dispatch(addTransactions({ transactions, accountId }))
+            : dispatch(updateTransactions({ transactions, accountId }));
+    }
+);
+
+const fetchTransactionStatus = createAsyncThunk(
+    `${SLICE_NAME}/fetchTransactionStatus`,
+    async ({ hash, signer_id, accountId }, { dispatch, getState }) => {
+        let status;
+        try {
+            const transactionDetails = await transactionExtraInfo({ hash, signer_id });
+            status = Object.keys(transactionDetails.status)[0];
+        } catch (error) {
+            status = 'notAvailable';
+        }
+        const checkStatus = ['SuccessValue', 'Failure'].includes(status);
+
+        const { actions: { updateTransactionStatus } } = transactionsSlice;
+        dispatch(updateTransactionStatus({ status, checkStatus, accountId, hash }));
+    }
+);
+
+const transactionsSlice = createSlice({
+    name: SLICE_NAME,
+    initialState,
+    reducers: {
+        addTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+            set(state, ['transactions', 'byAccountId', accountId], transactions);
+        },
+        updateTransactions(state, { payload }) {
+            const { transactions, accountId } = payload;
+
+            const transactionsState = state.transactions.byAccountId[accountId];
+            const hash = transactionsState.map((t) => t.hash_with_index);
+
+            transactions.forEach((t) => {
```

Comment:
> Thinking about it, I am assuming it's because we don't want to replace an existing entry that may have had it's tx status loaded for it already?

Correct. We also had automatic transactions refresh every few seconds, which was removed by mistake some time ago. There is an issue to add it again. 

Added comment.